### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by swainoe**

### CODE REVIEW

**Comment by swainoe** [comment]
merge pull request

File: lib/data/calcdelta.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+module.exports = function calcDelta (oldData, newData) {
+
+  var delta = {'delta': true};
+  var changesFound = false;
+
+  // if there's no updates done so far, just return the full set
+  if (!oldData.sgvs) { return newData; }
+
+  function nsArrayTreatments(oldArray, newArray) {
```
Comment (jasoncalabrese) [inline lib/data/calcdelta.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830940/issues/source?bid=2452374&fileBranchId=3244826#l13)

File: lib/data/calcdelta.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    return result;
+  }
+
+  function sort(values) {
+    values.sort(function sorter(a, b) {
+      return a.mills - b.mills;
+    });
+  }
+
+  function compressArrays(delta, newData) {
```
Comment (jasoncalabrese) [inline lib/data/calcdelta.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830940/issues/source?bid=2452374&fileBranchId=3244826#l86)

File: lib/data/calcdelta.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            sort(deltaData);
+            delta[a] = deltaData;
+          }
+        }
+      }
+    }
+    return {'delta': delta, 'changesFound': changesFound};
+  }
+
+  function deleteSkippables(delta,newData) {
```
Comment (jasoncalabrese) [inline lib/data/calcdelta.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830940/issues/source?bid=2452374&fileBranchId=3244826#l117)

File: lib/data/calcdelta.js (hunk 1 above)
Comment (jasoncalabrese) [inline lib/data/calcdelta.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830940/issues/source?bid=2452374&fileBranchId=3244826#l13)

File: lib/plugins/openaps.js [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+      } else if (moments.when && (noWarning || moments.when.isAfter(recent))) {
+        status.symbol = '◉';
+        status.code = 'waiting';
+        status.label = 'Waiting';
+      }
+
+      return status;
+    }
+
+    _.forEach(recentData, function eachStatus (status) {
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830972/issues/source?bid=2452374&fileBranchId=3244826#l105)

File: lib/plugins/openaps.js [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+          deviceInfo.push('@ ' + best[2] + 'dB');
+        }
+      }
+      events.push({
+        time: device.status.when
+        , value: deviceInfo.join(' ')
+      });
+    });
+
+    var sorted = _.sortBy(events, function toMill(event) {
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830972/issues/source?bid=2452374&fileBranchId=3244826#l288)

File: lib/plugins/openaps.js [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+        , message: 'Last Loop: ' + formatAgo(prop.lastLoopMoment, now.valueOf())
+        , pushoverSound: 'echo'
+        , plugin: openaps
+        , debug: prop
+      });
+    }
+  };
+
+  openaps.findOfflineMarker = function findOfflineMarker(sbx) {
+    return _.findLast(sbx.data.treatments, function match(treatment) {
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830972/issues/source?bid=2452374&fileBranchId=3244826#l195)

File: lib/plugins/openaps.js [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+
+    var prefs = openaps.getPrefs(sbx);
+
+    function valueString (prefix, value) {
+      return value ? prefix + value : '';
+    }
+
+    var events = [ ];
+
+    function addSuggestion() {
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830972/issues/source?bid=2452374&fileBranchId=3244826#l213)

File: lib/plugins/openaps.js [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+    sbx.offerProperty('openaps', function setOpenAPS ( ) {
+      return openaps.analyzeData(sbx);
+    });
+  };
+
+  openaps.analyzeData = function analyzeData (sbx) {
+    var recentHours = 6; //TODO dia*2
+    var recentMills = sbx.time - times.hours(recentHours).msecs;
+
+    var recentData = _.filter(sbx.data.devicestatus, function (status) {
```
Comment (jasoncalabrese) [inline lib/plugins/openaps.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2439830972/issues/source?bid=2452374&fileBranchId=3244826#l35)

**Pull request closed without merging**